src/reducers/accounts.js

```diff
@@ -15,5 +16,16 @@ export default (state = initialState, { type, payload }) => {
 }
 
 export function getAccount(state, address) {
-  return state.accounts[address] || null;
+  return state.accounts[address] || {};
+}
+
+export function getCurrentAccountForDisplay(state, methods = { getAccount, fromWei }) {
+  // redux-first-router has issues with '0x' strings
+  const locationAddress = state.location.payload.address || '';
+  const address = locationAddress.replace('_', '');
+  const accountData = methods.getAccount(state, address);
+  const balanceInWei = accountData.balance ? accountData.balance.toString(10) : '';
+  const balanceInEther = methods.fromWei(accountData.balance, 'ether');
```

Comment:
It would be consistent to check for `accountData.balance` both for `balanceInWei` and `balanceInEther`. Otherwise we'd have the Wei balance as `''` and Ether `0`. You could wrap both with regular if-statement.